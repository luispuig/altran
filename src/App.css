body { font-weight: 200 !important; }
h1, h2, h3, h4, h5, h6, .h1, .h2, .h3, .h4, .h5, .h6 { font-weight: 300 !important;}


#main {
	padding-top: 30px;
	padding-bottom: 30px;
}

.elevation-1 {
	box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
	transition: all 0.3s cubic-bezier(.25,.8,.25,1);
	border: none !important;
}

.elevation-1.elevation-hover:hover {
	box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
	border: none !important;
}

.elevation-2 {
	box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
	border: none !important;
}

.elevation-3 {
	box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23);
	border: none !important;
}

.elevation-4 {
	box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
	border: none !important;
}

.elevation-5 {
	box-shadow: 0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22);
	border: none !important;
}

.bg-primary {
	background-color: #42a5f5 !important;
}

.bg_color_main { background: #42a5f5 !important; }
.bg_color_light { background: #80d6ff !important;  }
.bg_color_dark { background: #0077c2 !important;   }

.text_color_main   { color: #42a5f5 !important;}
.text_color_light { color: #80d6ff !important;}
.text_color_dark { color: #0077c2 !important;}
