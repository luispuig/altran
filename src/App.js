// @flow
import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";

import Home from "./pages/Home/Home";
import About from "./pages/About/About";
import './App.css';

type _Props = {};

class App extends Component<_Props> {
  render() {
    return (
      <Router basename="/">
        <div>
          <header>
            <RouteLink to="/"      activeOnlyWhenExact={true} label="Home"/>
            <RouteLink to="/about" activeOnlyWhenExact={true} label="About"/>
          </header>
          <Route exact path="/" component={Home} />
          <Route path="/about" component={About} />
        </div>
      </Router>
    );
  }
}

type RouteLink_Props = {
  label: string,
  to: string,
  activeOnlyWhenExact?: boolean,
}


const RouteLink = ({ label, to, activeOnlyWhenExact }:RouteLink_Props) => (
    <Route path={to} exact={activeOnlyWhenExact} children={({ match }:{ match?:boolean }) => (
          <Link className='nav-link' to={to}>{label}</Link>
    )}/>
);


export default App;
