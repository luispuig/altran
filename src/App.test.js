import React from 'react';
import ReactDOM from 'react-dom';

import { Provider } from 'react-redux';
import { createMockStore } from 'redux-test-utils';

import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
import { shallowWithStore } from 'enzyme-redux';
import { shallowToJson } from 'enzyme-to-json';
configure({ adapter: new Adapter() });


import App from './App';

const initialData = { temperature: 0, state: 'request', error: undefined  };
const cities = [
  { id: 3871336, name: "Santiago",      ...initialData },
  { id: 3435910, name: "Buenos Aires",  ...initialData },
  { id: 3936456, name: "Lima",          ...initialData },
  { id: 3448439, name: "Sao Paulo",     ...initialData }
];

const testState = {
  cities,
  citiesTemperatures: []
};

const store = createMockStore(testState);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Provider store={store}><App /></Provider>, div);
});

it('match on Shallow mode', () => {
  const component = shallowWithStore(<App />, store);

  const tree = shallowToJson(component, {mode: 'deep'});
  expect(tree).toMatchSnapshot();

});
