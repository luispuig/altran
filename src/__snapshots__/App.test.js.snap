// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match on Shallow mode 1`] = `
<BrowserRouter
  basename="/"
>
  <div>
    <header>
      <RouteLink
        activeOnlyWhenExact={true}
        label="Home"
        to="/"
      />
      <RouteLink
        activeOnlyWhenExact={true}
        label="About"
        to="/about"
      />
    </header>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/about"
    />
  </div>
</BrowserRouter>
`;
